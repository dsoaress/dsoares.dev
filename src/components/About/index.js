import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'

import BlockContent from '../BlockContent'
import Social from '../Social'
import * as S from './styled'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <S.Wrapper>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <S.Grid>
        <S.TextWrapper>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </S.TextWrapper>
        <S.Image fluid={sanitySiteSettings.bio.image.asset.fluid} />
      </S.Grid>
    </S.Wrapper>
  )
}

export default About
