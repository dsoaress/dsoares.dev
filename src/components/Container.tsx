import cn from 'classnames'
import type { HTMLAttributes } from 'react'

export function Container({ className, ...props }: HTMLAttributes<HTMLDivElement>) {
  return <main className={cn('max-w-screen-xl px-6 mx-auto', className)} {...props} />
}
