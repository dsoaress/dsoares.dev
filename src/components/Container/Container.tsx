import cn from 'classnames'
import type { HTMLAttributes } from 'react'

export function Container({ className, ...props }: HTMLAttributes<HTMLDivElement>) {
  return <main className={cn('mx-auto max-w-screen-xl px-6', className)} {...props} />
}
