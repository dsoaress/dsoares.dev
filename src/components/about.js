import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

import BlockContent from './block-content'
import Social from './social'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <div>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <div>
        <div>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </div>
        <Img fluid={sanitySiteSettings.bio.image?.asset.fluid} />
      </div>
    </div>
  )
}

export default About
