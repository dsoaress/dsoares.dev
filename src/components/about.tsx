import { Block } from './block'

export function About() {
  return (
    <Block className="col-span-12 text-3xl leading-snug">
      <p className="text-neutral-400">
        Specializing in <span className="text-neutral-50">scalable, robust applications</span>, I
        focus on delivering seamless user experiences and optimizing backend systems for maximum
        efficiency.
      </p>
    </Block>
  )
}
