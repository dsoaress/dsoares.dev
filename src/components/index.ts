export * from './about'
export * from './block'
export * from './footer'
export * from './header'
export * from './home-container'
export * from './logo'
export * from './social'
