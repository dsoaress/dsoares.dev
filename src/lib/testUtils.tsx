import { render } from '@testing-library/react'
import { ComponentType, ReactElement, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles/theme'

type ProvidersProps = {
  children?: ReactNode
}

const Providers = ({ children }: ProvidersProps) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
