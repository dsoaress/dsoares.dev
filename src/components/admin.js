import { useState } from 'react'
import useSWR, { mutate } from 'swr'
import { signOut } from 'next-auth/client'

import Spinner from '@/components/spinner'
import { fetcher } from '@/lib/fetcher'

export default function Tasks() {
  const [label, setLabel] = useState('')
  const [url, setUrl] = useState('')

  const { data } = useSWR('/api', fetcher, {
    refreshInterval: 500
  })

  if (!data) return <Spinner />

  async function handleSubmit(e) {
    e.preventDefault()
    mutate('/api', { ...data, data: { links: { label, url } } }, false)

    await fetch('/api', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ link: { label, url } })
    })

    mutate('/api')
    setLabel('')
    setUrl('')
  }

  return (
    <div>
      <div>
        <button onClick={() => signOut()}>Sign out</button>
      </div>
      <form onSubmit={handleSubmit}>
        <div className="text-black">
          <label>Label</label>
          <input
            type="text"
            value={label}
            onChange={e => setLabel(e.target.value)}
            required
          />
          <label>Url</label>
          <input
            type="text"
            value={url}
            onChange={e => setUrl(e.target.value)}
            required
          />
        </div>
        <button type="submit">Add</button>
      </form>
      <div>
        {!data.links || data.links.length === 0 ? (
          <p>
            It looks like you don't have any more tasks... Would you like to add
            a new one?
          </p>
        ) : (
          data.links.map(link => (
            <div key={link._id}>
              <p>{link.label}</p>
              <p>{link.url}</p>
            </div>
          ))
        )}
      </div>
    </div>
  )
}
