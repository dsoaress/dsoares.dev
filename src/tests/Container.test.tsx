import { Container } from '@/components/Container'
import { render, screen } from '@/lib/testUtils'

describe('<Container />', () => {
  it('Renders with children', () => {
    render(<Container>Children</Container>)

    expect(screen.getByText('Children')).toBeInTheDocument()
    expect(screen.getByText('Children')).toMatchSnapshot()
  })
})
