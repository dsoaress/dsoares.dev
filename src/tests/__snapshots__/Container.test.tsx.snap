// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Renders with children 1`] = `
.c0 {
  max-width: 1280px;
  padding: 0 1.5rem;
  margin: auto;
}

<main
  class="c0"
>
  Children
</main>
`;
